### DISCUSSION TITLE
v3.8.9 tag

### DISCUSSION THREAD

**Discussion by inikulshin** [rabbitmq/rabbitmq-ct-helpers#50] [General]
@pjk25 , why there are no new tags in this repo since 3.7?

After breaking https://github.com/rabbitmq/rabbitmq-ct-helpers/commit/37b5167f9e99c3090cf73d7bf840d298915b13d2 it is impossible to build custom plugins with **v3.8.9**. `make deps` brings `master` instead of non-existing tag `v3.8.9`, and fails with 

ERLC   rabbit_control_helper.erl rabbit_ct_broker_helpers.erl rabbit_ct_config_schema.erl rabbit_ct_helpers.erl rabbit_ct_proper_helpers.erl rabbit_ct_vm_helpers.erl rabbit_mgmt_test_util.erl
**src/rabbit_ct_helpers.erl:12: badly formed deprecated attribute** {is_mixed_versions,1,[85,115,101,32,105,115,95,109,105,120,101,100,95,118,101,114,115,105,111,110,115,47,48,32,105,110,115,116,101,97,100]}
erlang.mk:5452: recipe for target 'ebin/rabbitmq_ct_helpers.app' failed

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
This repository is not tagged and is not a part of RabbitMQ distribution. It's just a bunch of test helpers that rarely change.

There is a [3.8 branch](https://github.com/rabbitmq/rabbitmq-ct-helpers/tree/v3.8.x) you can use with `3.8.9`.

**Comment by michaelklishin** [comment]
My best guess is that

 * These test dependencies are not pinned to the same branch
 * `-deprecated` is a relatively new module attribute that your Erlang version does not support. `3.8.9` is almost 10 months old but it [does support Erlang 23](https://www.rabbitmq.com/which-erlang.html), which does support this attribute (I can compile `rabbitmq_ct_helpers` using it just fine)

So there is no need to tag this repository. You need to pin the dependency to the `v3.8.x` branch in `rabbitmq-components.mk` and/or use Erlang 23. Or just use the binary packages produced for you, all the way to `3.8.18`.

**Reply by HoloRin** [reply]
That is correct. rabbitmq-ct-helpers (and rabbitmq-ct-client-helpers) are not tagged, but are expected to match the branch of rabbitmq-server. rabbitmq-components.mk should pull matching branches (https://github.com/rabbitmq/rabbitmq-server/blob/1068e3eda4bc43168c4053dd60be9ddafcf584c1/rabbitmq-components.mk#L57-L58), but in the case of a custom plugin, this may not work as expected.

**Reply by inikulshin** [reply]
Thanks,

So, is there a workaround?

I'm trying to build custom plugins (in **docker**, but I don't think it matters).

Ubuntu 16.04
Erlang **22.1**

	// copy all my files (sources, makefiles, ...) to .

	git init \
	&& git config user.name "someone" \
	&& git config user.email "someone@someplace.com" \
	&& git checkout -b v3.8.9 \
	&& git commit --allow-empty --allow-empty-message -m "" \
	&& make deps test-deps || (mix hex.config unsafe_https true && make deps test-deps)

Can I change something in git commands to cause make to get rabbitmq-ct-helpers **v3.8.x** rather than **master**?

**Reply by HoloRin** [reply]
I just tried the following:
1. `git checkout v3.8.x` in rabbitmq-server
2. `make -C deps/rabbit test-deps current_rmq_ref=v3.8.x`

This pulled `v3.8.x` for the helpers

I also tried:
- `make -C deps/rabbit test-deps current_rmq_ref=v3.8.9 base_rmq_ref=v3.8.x` which reported `v3.8.9` of the helpers in the output, but seemed to actually fetch `v3.8.9` as well.

So, it may still depend on your layout, and which version of `rabbitmq-components.mk` that you have, but the `base_rmq_ref` and `current_rmq_ref` make variables seem to work fine for overriding the value that seems to be detected incorrectly.

**Reply by inikulshin** [reply]
@pjk25 thanks a lot! 

`base_rmq_ref=v3.8.x` is exactly what I needed.
